lib/inc/facter/facts/fact.hpp

```diff
@@ -426,6 +426,26 @@ namespace facter { namespace facts {
          */
         constexpr static char const* macosx_productversion_minor = "macosx_productversion_minor";
 
+        /**
+        * The fact for Windows to specify if is Server or Desktop Edition variant 
+        */
+        constexpr static char const* windows_edition_id = "edition_id";
```

Comment:
I'm not sure why we want these as top-level facts. Why list them in two places?

Reply:
Not sure what you mean @MikaelSmith about listing them in two places ?

Reply:
It looks like you can get this fact (for example) via `facter edition_id` or `facter windows.edition_id`.

Reply:
Ok - Understand you now - they should really only be readable as under ```windows.```

Reply:
Hmm - did a test there and got the following:
```
PS C:\Users\Administrator\pscconfeu-dsc> facter edition_id
ServerStandard
PS C:\Users\Administrator\pscconfeu-dsc> facter windows.edition_id

PS C:\Users\Administrator\pscconfeu-dsc> facter windows.system32

PS C:\Users\Administrator\pscconfeu-dsc> facter system32
C:\Windows\system32
PS C:\Users\Administrator\pscconfeu-dsc>
```

So both ```edition_id``` and ```system32``` are presented as top level facts.

Reply:
Oh, dot-indexing may not actually work on the CLI. I thought that had been added. What's present when you run `facter windows`?

Reply:
It returns nothing.

Reply:
Some more output:
```
PS C:\Users\Administrator\pscconfeu-dsc> facter os.windows
{
  edition_id => "ServerStandard",
  installation_type => "Server",
  product_name => "Windows Server 2016 Standard",
  release_id => "1607",
  system32 => "C:\Windows\system32"
}
PS C:\Users\Administrator\pscconfeu-dsc> facter os.windows.edition_id
ServerStandard
PS C:\Users\Administrator\pscconfeu-dsc>
```

Reply:
So dot indexing does work, just forgot to reference from the top ```os```
I would tend to agree with you though that the top level facts should be removed.

Reply:
> or example) via `facter edition_id` or `facter windows.edition_id`.

hi @MikaelSmith this functionality is the same for all the facts provided by facter. For example on OSX:
```
mihai:puppet mihai.buzgau$ facter macosx_buildversion
18A391 
```
returns the same thing as:
```
mihai:puppet mihai.buzgau$ facter os.macosx.build
18A391 
```

If this is not the intended functionality, it should be tacked in a different task

Reply:
I have also renamed the top-level fact to windows_edition_id in order to be seen as a windows specific fact.

Reply:
Facter originally only used top-level facts. Around Facter 2.5/3 we introduced the concept of structured facts, and recreated structured versions of many top-level facts. The goal has been to primarily introduce new facts as part of structured versions.

You may notice that if you run `facter` you only see `os.macosx.build` and not `macosx_buildversion`; if you run with `facter --show-legacy` it shows both.

This isn't something to fix generally, it was a change in design that supported backwards compatibility for using old ways of referencing facts.

Reply:
Looks like this has been merged.

I have just verified the latest build and using the ```--show-legacy``` option that @MikaelSmith mentions gives the following output:
```
{... Stuff redacted...}
operatingsystem => windows
operatingsystemmajrelease => 2016
operatingsystemrelease => 2016
os => {
  architecture => "x64",
  family => "windows",
  hardware => "x86_64",
  name => "windows",
  release => {
    full => "2016",
    major => "2016"
  },
  windows => {
    edition_id => "ServerStandard",
    installation_type => "Server",
    product_name => "Windows Server 2016 Standard",
    release_id => "1607",
    system32 => "C:\Windows\system32"
  }
}
osfamily => windows
{... Stuff redacted...}
uuid => FB761A42-26C0-6855-0683-C8B80E040BD7
virtual => vmware
windows_edition_id => ServerStandard
windows_installation_type => Server
windows_product_name => Windows Server 2016 Standard
windows_release_id => 1607
```